src/porepy/applications/verification_setups/mandel_biot.py

```diff
@@ -0,0 +1,1569 @@
+
+        t = self.time_manager.time
+        uy_north_bc = self.exact_sol.vertical_displacement_profile(yf_north, t)
+        bc_vals[1::2][sides.north] = uy_north_bc
+
+        return bc_vals
+
+
+class MandelBoundaryConditionsSinglePhaseFlow(mass.BoundaryConditionsSinglePhaseFlow):
+
+    domain_boundary_sides: Callable[[pp.Grid], pp.bounding_box.DomainSides]
+    """Utility method to access the indices of the sides of the domain."""
+
+    def bc_type_darcy(self, sd: pp.Grid) -> pp.BoundaryCondition:
+        """Define boundary condition types for the flow subproblem.
+
+        All sides are set as Neumann, except the East side, which is Dirichlet.
+
+        Parameters:
+            sd: Subdomain grid.
+
+        Returns:
+            Scalar boundary condition representation.
+
+        """
+        # Inherit from base class. All sides are set as Dirichlet.
+        bc = super().bc_type_darcy(sd)
+
+        sides = self.domain_boundary_sides(sd)
+
+        # West side
+        bc.is_neu[sides.west] = True
+        bc.is_dir[sides.west] = False
+
+        # North side
+        bc.is_neu[sides.north] = True
+        bc.is_dir[sides.north] = False
+
+        # South side
+        bc.is_neu[sides.south] = True
+        bc.is_dir[sides.south] = False
+
+        return bc
+
+
+class MandelPoromechanicsBoundaryConditions(
+    MandelBoundaryConditionsSinglePhaseFlow,
+    MandelBoundaryConditionsMechanicsTimeDependent,
+):
+    """Mixer class for poromechanics boundary conditions."""
+
+
+# -----> Solution strategy
+class MandelSolutionStrategy(poromechanics.SolutionStrategyPoromechanics):
+    """Solution strategy for Mandel's problem."""
+
+    exact_sol: MandelExactSolution
+    """Exact solution object."""
+
+    plot_results: Callable
+    """Method that plots pressure, displacement, flux, force, and degree of
+    consolidation in non-dimensional forms.
+
+    """
+
+    results: list[MandelSaveData]
+    """List of :class:~`MandelSaveData` objects, containing the results of the
+    verification.
+
+    """
+
+    def __init__(self, params: dict) -> None:
+        """Constructor of the class.
+
+        Parameters:
+            params: Parameters of the verification setup.
+
+        """
+        super().__init__(params)
+
+        self.exact_sol: MandelExactSolution
+        """Exact solution object."""
+
+        self.results: list[MandelSaveData] = []
+        """List of stored results from the verification."""
+
+    def set_materials(self):
+        """Set material parameters.
+
+        Add exact solution object to the simulation model after materials have been set.
+
+        """
+        super().set_materials()
+        self.exact_sol = MandelExactSolution(self)
+
+        # Biot's coefficient must be one
+        assert self.solid.biot_coefficient() == 1
+
+    def initial_condition(self) -> None:
+        """Set initial conditions.
+
+        Initial conditions are given by Eqs. (41) - (43) from [1].
+
+        References:
```

Comment:
Should suffice to define references once.

Reply:
Fixed.